crates/duty-tracker/src/contract_manager.rs

```diff
@@ -463,7 +493,166 @@ impl ContractManagerCtx {
         }
     }
 
-    fn execute_duty(&mut self, _duty: OperatorDuty) {
-        todo!() // execute duty
+    async fn execute_duty(&mut self, duty: OperatorDuty) {
+        match duty {
+            OperatorDuty::PublishWOTSKeys { txid } => {
+                let operator_pk = self
+                    .s2_client
+                    .general_wallet_signer()
+                    .pubkey()
+                    .await
+                    .unwrap();
+                let wots_client = self.s2_client.wots_signer();
+                /// VOUT is static because irrelevant so we're just gonna use 0
+                const VOUT: u32 = 0;
+                // withdrawal_fulfillment uses index 0
+                let withdrawal_fulfillment = Wots256PublicKey::from_flattened_bytes(
+                    &wots_client.get_256_key(txid, VOUT, 0).await.unwrap(),
+                );
+                const NUM_FQS: usize = NUM_U256;
+                const NUM_PUB_INPUTS: usize = NUM_PUBS;
+                const NUM_HASHES: usize = NUM_HASH;
+                let public_inputs_ftrs: [_; NUM_PUB_INPUTS] =
+                    std::array::from_fn(|i| wots_client.get_256_key(txid, VOUT, i as u32));
+                let fqs_ftrs: [_; NUM_FQS] = std::array::from_fn(|i| {
+                    wots_client.get_256_key(txid, VOUT, (i + NUM_PUB_INPUTS) as u32)
+                });
+                let hashes_ftrs: [_; NUM_HASHES] =
+                    std::array::from_fn(|i| wots_client.get_128_key(txid, VOUT, i as u32));
```

Comment:
Are you sure that these are the **public key** and not the **private** key?
I am feeling that the bytes that comes from the secret service are the **private** key ones.